mobile/test/non_hermetic/gcp_traffic_director_integration_test.cc

```diff
@@ -0,0 +1,133 @@
+#include "gtest/gtest.h"
+#include "library/common/data/utility.h"
+#include "library/common/engine_handle.h"
+#include "library/common/types/c_types.h"
+
+namespace Envoy {
+namespace {
+
+using ::Envoy::Grpc::SotwOrDelta;
+using ::Envoy::Network::Address::IpVersion;
+
+// The One-Platform API endpoint for Traffic Director.
+constexpr char TD_API_ENDPOINT[] = "trafficdirector.googleapis.com";
+// The name of the project in Google Cloud Console; copied from the project_id
+// field in the generated JWT token.
+constexpr char PROJECT_NAME[] = "td-testing-gfq";
+// The project number of the project, found on the main page of the project in
+// Google Cloud Console.
+constexpr char PROJECT_ID[] = "798832730858";
+// Copied from the "client_id" field in the generated JWT token.
+constexpr char CLIENT_ID[] = "102524055118681734203";
+// Copied from the "private_key_id" field in the generated JWT token.
+constexpr char PRIVATE_KEY_ID[] = "e07f02d49044a533cf4342d138eacecc6acdb6ed";
+
+// Using a JWT token to authenticate to Traffic Director.
+std::string jwtToken() {
+  const std::string email =
+      absl::Substitute("$0-compute@developer.gserviceaccount.com", PROJECT_ID);
+  const std::string cert_url = absl::Substitute("https://www.googleapis.com/robot/v1/metadata/x509/"
+                                                "$0-compute%40developer.gserviceaccount.com",
+                                                PROJECT_ID);
+  const std::string private_key(TestEnvironment::getCheckedEnvVar("GCP_JWT_PRIVATE_KEY"));
+
+  return absl::Substitute(
+      R"json({
+        "private_key": "$0",
+        "private_key_id": "$1",
+        "project_id": "$2",
+        "client_email": "$3",
+        "client_id": "$4",
+        "client_x509_cert_url": "$5",
+        "type": "service_account",
+        "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+        "token_uri": "https://oauth2.googleapis.com/token",
+        "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs"
+      })json",
+      private_key, PRIVATE_KEY_ID, PROJECT_NAME, email, CLIENT_ID, cert_url);
+}
+
+// Tests that Envoy Mobile can connect to Traffic Director (an xDS management server offered by GCP)
+// via a test GCP project, and can pull down xDS config for the given project.
+class GcpTrafficDirectorIntegrationTest
+    : public Envoy::BaseClientIntegrationTest,
+      public testing::TestWithParam<std::tuple<IpVersion, SotwOrDelta>> {
+public:
+  GcpTrafficDirectorIntegrationTest() : BaseClientIntegrationTest(ip_version()) {
+    // Force register the Google gRPC library.
+    Grpc::forceRegisterDefaultGoogleGrpcCredentialsFactory();
+    // Force register the gRPC mux implementations.
+    Config::forceRegisterGrpcMuxFactory();
+    Config::forceRegisterNewGrpcMuxFactory();
+    Config::forceRegisterAdsConfigSubscriptionFactory();
+    // Force register the cluster factories used by the test.
+    Upstream::forceRegisterEdsClusterFactory();
```

Comment:
optional: tracking bug for config option (as we talked about) and TODO here so it doesn't get missed

honestly you'll remember it so really optional =P